samples/correlation-csharp/Readme.md

```diff
@@ -0,0 +1,57 @@
+
+# Sample Scenario
+
+Refer to the scenario that includes the samples. Some scenarios do not support distributed tracing. We'll support it in the future. For executing samples, call `HttpStart_*` functions as the endpoints. 
+
+| Scenario | Description | Endpoint | Supported | comment |
+| ------- | ------------------- | -------- | --------- | ------- |
+| Simple Orchestration | A simple Orchestration with an Activity | /api/HttpStart_SimpleOrchestration | yes | |
+| Multiple LayerOrchestration With Retry | A complex orchestration that includes sub orchestration, errors, and retry | /api/HttpStart_MultiLayerOrchestrationWithRetry | yes | |
+| Correlation integration | Getting a tracing Id from outside of the function, also getting tracing Id in the orchestrator and activity | /api/HttpStart_AnExternalSystem | yes | The endpoint represents an external sytem, it call a function that start an orchestration |
+| Fan Out Fan In | Simple [Fan Out/Fan In](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-overview?tabs=csharp#fan-in-out) Pattern implementation | /api/HttpStart_FanOutFanIn |
+| Durable Entity Orchestration | [Stateful Entities Pattern](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-orchestrations?tabs=csharp) implementation | /api/HttpStart_EntityCounter |no | Entity has not supporeted yet, however, it correlate for the other parts |
+| HTTP endpoints | [Calling HTTP endpoints](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-orchestrations?tabs=csharp) implementation | /api/HttpStart_HttpEndpoints | no | |
+| Human Interaction | [Human Interaction](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-overview?tabs=csharp#human) pattern implementation | /api/HttpStart_ExternalEvent | no | After staring the orchestration, call /api/ApprovalOne and /api/ApprovalTwo with `{"Message": "Approved!" , "InstanceId":"<YOUR_INSTANCE_ID>"}` with POST |
+| ContinueAsNew | [Eternal Orchestrator](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-eternal-orchestrations?tabs=csharp) implementation| /api/HttpStart_ExternalOrchestrations | yes |  |
+| Monitoring | [Monitor ](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-overview?tabs=csharp#monitoring) Pattern implementation | /api/HttpStart_Monitor?JobId={JobId} | no | You can see all the telemetries on the End to End transaction as the same Operation ID telemetry. However, the parent-children relationship sometimes missing. |
+
+# Limitation
+
+# FAQ
+
+### 1. In some supported Scenario, the correlation relationship looks broken
+
+The Distributed Tracing system track the Orchestration Request Telemetry after the completion of the orchestrator. Wait for a while, then look at the telemetry again. 
```

Comment:
Yes. It is inherent limitation.  The Application Insights has a delay that we can watch the telemetry on portal. Also, this distributed tracing system emit telemetry once an orchestration has been finished. So that a request telemetry of an orchestrator comes last.